app/uk/gov/hmrc/agentassurance/auth/AuthActions.scala

```diff
@@ -85,7 +86,7 @@ trait AuthActions extends AuthorisedFunctions {
     }
   }
 
-  def withAffinityGroupAgent(action: Request[AnyContent] => Future[Result]) = Action.async {
+  def withAffinityGroupAgent(action: Request[AnyContent] => Future[Result])(implicit ec: ExecutionContext): Action[AnyContent] = Action.async {
```

Comment:
in every method, we duplicated implicit ec declaration and headerCarrier definition, we can fix these ?

Reply:
👍 on it!